### DISCUSSION TITLE
Jalali - Persian - Shamsi date template

### DISCUSSION THREAD

**Discussion by behdadmansouri** [SilentVoid13/Templater#640] [Templates Showcase]
# What does it do?
Adds today's Jalali - Persian date  - Shamsi(with calendar link)
### Template:
```
[[<% tp.date.now("YYYY-MM-DD") + "|" + new Intl.DateTimeFormat('en-u-ca-persian', { dateStyle: 'full' }).format(new Date)%>]]
```
# output
```[[2022-05-20|Friday, Ordibehesht 30, 1401 AP]]```
# Ideas 💡
### It can be in Farsi
By changing "en-u-ca-persian" to "fa-u-ca-persian"
```[[2022-05-20|۱۴۰۱ اردیبهشت ۳۰, جمعه]]```
### Can be used for different dates other than "now" (like "creation date")
- creation date is accessed via `tp.file.creation_date("YYYYMMDDHHSS")`
### Can be added to the frontmatter
- [Via this guide](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/frontmatter-module.html)
### The date format can be changed
- The order of "weekday, month" can be changed
- [Code Project Reference](https://reference.codeproject.com/javascript/reference/global_objects/datetimeformat)
### Islamic - Hijri - Arabic calendar
- By changing "en-u-ca-persian" to "en-u-ca-islamic"
### Any other calendar, really
nu
    Numbering system. Possible values include: "arab", "arabext", "bali", "beng", "deva", "fullwide", "gujr", "guru", "hanidec", "khmr", "knda", "laoo", "latn", "limb", "mlym", "mong", "mymr", "orya", "tamldec", "telu", "thai", "tibt".
ca
    Calendar. Possible values include: "buddhist", "chinese", "coptic", "ethioaa", "ethiopic", "gregory", "hebrew", "indian", "islamic", "islamicc", "iso8601", "japanese", "persian", "roc". 
[Code Project Reference](https://reference.codeproject.com/javascript/reference/global_objects/datetimeformat)

All the Code Project references are also available in [Mozilla's website](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat) but I find the docs to be dry

I found out about this neat function [from this stack overflow post](https://stackoverflow.com/questions/47669912/convert-gregorian-date-to-jalali)

**Comment by Archie-2021** [comment]
> 

Very cool, I also made some aliases for my daily notes using it,

```
<%* 
const dateVar = tp.file.title
const outVar = new Intl.DateTimeFormat('en-u-ca-persian', { year: "2-digit" ,month: "2-digit", day: "2-digit" }).format(new Date(dateVar))
tR+= outVar
%>
```

**Comment by parvizshams** [comment]
Hi Thank you for your useful guide
Is there any way to create periodic note with shamsi date Name instead of gregorian date in obsidian ?!

**Comment by parvizshams** [comment]
و آیا راهی برای نمایش تاریخ شمسی بصورت صحیح وجود دارد
بجای ۱۴۰۱ خرداد ۲۲, یکشنبه
نوشته شود : یکشنبه، ۲۲ خرداد ۱۴۰۱

**Reply by Archie-2021** [reply]
> و آیا راهی برای نمایش تاریخ شمسی بصورت صحیح وجود دارد بجای ۱۴۰۱ خرداد ۲۲, یکشنبه نوشته شود : یکشنبه، ۲۲ خرداد ۱۴۰۱

Very unlikely, it is only possible to convert that output by modifying the string and I have no idea how to use regex on Farsi alphabet, it was very easy if it was in Latin

**Reply by parvizshams** [reply]
Ok, Thank you
And do you have any idea for my other question I have mentioned above?
Shamsi date for periodic daily note name

**Reply by Archie-2021** [reply]
> Ok, Thank you And do you have any idea for my other question I have mentioned above? Shamsi date for periodic daily note name

I don't think it is possible, it is in the settings of periodic notes and only accepts a very limited format. You can ask it as a feature request to them, you can also rename the notes automatically using Templater, very easily, but then daily notes/periodic notes plugin would not detect them and it is useless if you need to use daily notes features like calendar and such and not just having notes without anything else.

**Reply by parvizshams** [reply]
Ok. Thank you very much!
You are very helpful
And as a last question, is there any way to have both RTL and LTR blocks of text in the same document

**Reply by AB1908** [reply]
I think that's more of an Obsidian question. Perhaps ask in the Discord server?

**Reply by Archie-2021** [reply]
> Ok. Thank you very much! You are very helpful And as a last question, is there any way to have both RTL and LTR blocks of text in the same document

np,

yes you can have them using html tags like 
```
<div dir="rtl">
</div> 
```

It is hard to write, especially if typing English words in the middle of it ,as it is only doing right alignment not the full RTL, but it is working at the end. I have not following closely on this issue for a while but there are some good threads on the obsidian forums like this [RTL Support - Feature requests - Obsidian Forum](https://forum.Obsidian.Md/t/rtl-support/1438/58) on this issue and how to work around it

**Reply by parvizshams** [reply]
Thanks a Billion!

**Comment by Mahgozar** [comment]
thank you for this
